pkg/dataplane/util/ansibleee.go

```diff
@@ -78,12 +80,17 @@ func (a *EEJob) JobForOpenStackAnsibleEE(h *helper.Helper) (*batchv1.Job, error)
 
 	args := a.Args
 
-	playbook := a.Playbook
+	artifact := a.Playbook
+	param := "-p"
 	if len(args) == 0 {
-		if len(playbook) == 0 {
-			playbook = CustomPlaybook
+		if len(a.PlaybookContents) > 0 {
```

Comment:
I think this changes the old behavior. Earlier `Playbook` provided used to supersede `PlaybookContents`. Now it's the reverse and `role` provided supersedes both of them. Is that what we want now?

Reply:
`playbook`, `playbookContents` and `role` are not mandatory fields, so we need to check which one isn't empty. Ideally, we shouldn't supersede, but it would happen when more than one field is filled. I think we might need a validation to ensure only one of the three fields is filed

Reply:
Yeah either have a validation or ensure that there is proper prededence. playbook->playbookContent->role

Reply:
I've updated the precedence